test/distributed/org/apache/cassandra/distributed/test/sai/SAIUtil.java

```diff
@@ -43,23 +43,85 @@ public class SAIUtil
+     */
+    @Deprecated
+    public static void waitForIndexQueryable(Cluster cluster, String keyspace, String index)
+    {
+        waitForIndexQueryable(cluster, keyspace, index, 1);
+    }
+
+    /**
+     * Waits until all indexes in the given keyspace become queryable, as verified by the specified coordinator node.
+     * This method checks index status from the coordinator's perspective, ensuring that when the coordinator
+     * executes queries, it will see the indexes as queryable.
+     * <p>
+     * Use this method when your test will query from a specific node, to ensure that the coordinator node
+     * has received gossip updates about index status.
+     *
+     * @param cluster the cluster
+     * @param keyspace the keyspace name
+     * @param coordinatorNode the node number (1-based) that will act as coordinator for queries
+     */
+    public static void waitForIndexQueryable(Cluster cluster, String keyspace, int coordinatorNode)
     {
         assertGossipEnabled(cluster);
         final List<String> indexes = getIndexes(cluster, keyspace);
         await().atMost(60, TimeUnit.SECONDS)
-               .untilAsserted(() -> assertIndexesQueryable(cluster, keyspace, indexes));
+               .untilAsserted(() -> assertIndexesQueryable(cluster, keyspace, indexes, coordinatorNode));
     }
 
     /**
-     * Waits until given index becomes queryable.
+     * Waits until the given index becomes queryable, as verified by the specified coordinator node.
+     *
+     * @param cluster the cluster
+     * @param keyspace the keyspace name
+     * @param index the index name
+     * @param coordinatorNode the node number (1-based) that will act as coordinator for queries
      */
-    public static void waitForIndexQueryable(Cluster cluster, String keyspace, String index)
+    public static void waitForIndexQueryable(Cluster cluster, String keyspace, String index, int coordinatorNode)
     {
         assertGossipEnabled(cluster);
         await().atMost(60, TimeUnit.SECONDS)
-               .untilAsserted(() -> assertIndexQueryable(cluster, keyspace, index));
+               .untilAsserted(() -> assertIndexQueryable(cluster, keyspace, index, coordinatorNode));
+    }
+
+    /**
+     * Waits until all indexes in the given keyspace become queryable on ALL cluster nodes.
+     * This is useful when tests query from multiple coordinators, ensuring that each node
+     * has received gossip updates about index status before queries are executed.
+     *
+     * @param cluster the cluster
+     * @param keyspace the keyspace name
+     */
+    public static void waitForIndexQueryableOnAllNodes(Cluster cluster, String keyspace)
+    {
+        assertGossipEnabled(cluster);
+        final List<String> indexes = getIndexes(cluster, keyspace);
+
+        for (int nodeNum = 1; nodeNum <= cluster.size(); nodeNum++)
+        {
+            final int node = nodeNum;
+            await().atMost(60, TimeUnit.SECONDS)
```

Comment:
Nit: we already have the same 60 seconds timeout for the index wait repeated three times; maybe we should put it in a constant?

Reply:
Fixed